### PR TITLE
List happs with all needed details and new holo cli

### PR DESCRIPTION
**Pull request by zo-el**
New CLI `holo`
```
user@nixos: holo help

usage: holo <OPTIONS>
   list-hosted-happs | l: List all the hosted happs registered on HHA
   help | h: display all usage options
```

### CODE REVIEW

File: overlays/holo-nixpkgs/holo-cli/default.nix [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
 
-  preConfigure = ''
-    cp -Lr ${npmToNix { inherit src; }} node_modules
-    chmod -R +w node_modules
-    patchShebangs node_modules
-  '';
+    installPhase = ''
+      install -Dm 755 holo.sh $out/bin/${name}
+      wrapProgram $out/bin/${name} \
+      --prefix PATH : ${makeBinPath [ jq git perl ]}
```
Comment (peeech) [inline overlays/holo-nixpkgs/holo-cli/default.nix:16]:
```suggestion
      --prefix PATH : ${makeBinPath [ jq hpos-holochain-client ]}
```

Comment (peeech) [reply overlays/holo-nixpkgs/holo-cli/default.nix:16]: IMHO git and perl are not needed here. For a change `hpos-holochain-client` is imported via `environment.systemPackages` but in the future might not be. Therefore adding `hpos-holochain-client`  to BinPath here will make this script future-proof.

File: profiles/logical/hpos/default.nix [hunk 2]
```diff
@@ -51,7 +51,7 @@ in
   # REVIEW: `true` breaks gtk+ builds (cairo dependency)
   environment.noXlibs = false;
 
-  environment.systemPackages = with holochainAllBinariesWithDeps.hpos; [ git hc-state lair-shim hpos-admin-client hpos-holochain-client hpos-reset hpos-update-cli holochain hc kitsune-p2p-proxy ];
+  environment.systemPackages = with holochainAllBinariesWithDeps.hpos; [ git hc-state lair-shim hpos-admin-client hpos-holochain-client hpos-reset hpos-update-cli holo-cli holochain hc kitsune-p2p-proxy ];
```
Comment (peeech) [inline profiles/logical/hpos/default.nix:54]:
```suggestion
  environment.systemPackages = with holochainAllBinariesWithDeps.hpos; [
    git
    hc-state
    lair-shim
    hpos-admin-client
    hpos-holochain-client
    hpos-reset
    hpos-update-cli
    holo-cli
    holochain
    hc
    kitsune-p2p-proxy
  ];
```
seems more readible

Comment (peeech) [reply profiles/logical/hpos/default.nix:54]: I love simplicity of this solution and how much value it adds at the same time. Well done!

**Review by peeech — CHANGES_REQUESTED** [review]

File: overlays/holo-nixpkgs/holo-cli/default.nix [hunk 3]
```diff
@@ -1,31 +1,23 @@
-{ stdenv, fetchFromGitHub, gitignoreSource, nodejs, npmToNix }:
+{ stdenv, makeWrapper, gitignoreSource, jq, perl, git }:
```
Comment (peeech) [inline overlays/holo-nixpkgs/holo-cli/default.nix:1]:
```suggestion
{ stdenv, makeWrapper, gitignoreSource, jq, pkgs }:
```

File: overlays/holo-nixpkgs/holo-cli/default.nix [hunk 4]
```diff
@@ -1,31 +1,23 @@
...
 
-  preConfigure = ''
-    cp -Lr ${npmToNix { inherit src; }} node_modules
-    chmod -R +w node_modules
-    patchShebangs node_modules
-  '';
+    installPhase = ''
+      install -Dm 755 holo.sh $out/bin/${name}
+      wrapProgram $out/bin/${name} \
+      --prefix PATH : ${makeBinPath [ jq hpos-holochain-client ]}
```
Comment (peeech) [inline overlays/holo-nixpkgs/holo-cli/default.nix:16]:
```suggestion
      --prefix PATH : ${makeBinPath [ jq pkgs.hpos-holochain-client ]}
```

File: overlays/holo-nixpkgs/holo-cli/default.nix (hunk 3 above)
Comment (alastairong) [reply overlays/holo-nixpkgs/holo-cli/default.nix:1]: Why do we want to pass all packages to the derivation?

File: overlays/holo-nixpkgs/holo-cli/default.nix (hunk 4 above)
Comment (peeech) [reply overlays/holo-nixpkgs/holo-cli/default.nix:16]: that should do it :-) @zo-el

File: overlays/holo-nixpkgs/holo-cli/default.nix (hunk 3 above)
Comment (peeech) [reply overlays/holo-nixpkgs/holo-cli/default.nix:1]: `pkgs.hpos-holochain-client`

**Review by peeech — APPROVED** [review]
well done!

**Pull request merged**